aws/cloud-init/master.script.tpl

```diff
@@ -189,57 +184,47 @@ EOF
   fi
 
 
-  # Provisioning of a SAML identity provider.
-  SAML_NAME=`aws ssm get-parameter --name /telekube/${cluster_name}/saml/name --region $EC2_REGION --query 'Parameter.Value' --output text --with-decryption`
-  SAML_ADMIN_GROUP=`aws ssm get-parameter --name /telekube/${cluster_name}/saml/admin-group --region $EC2_REGION --query 'Parameter.Value' --output text --with-decryption`
-  SAML_ENTITY_DESCRIPTOR=`aws ssm get-parameter --name /telekube/${cluster_name}/saml/entity-descriptor --region $EC2_REGION --query 'Parameter.Value' --output text --with-decryption`
-  if [ ! -z "$${SAML_ENTITY_DESCRIPTOR}" ]; then
-    cat <<EOF > saml.yaml
-kind: saml
-version: v2
-metadata:
-  name: sso
-spec:
-  acs: https://${cluster_name}/portalapi/v1/saml/callback
-  attributes_to_roles:
-  - name: groups
-    value: $${SAML_ADMIN_GROUP}
-    roles:
-    - '@teleadmin'
-  display: $${SAML_NAME}
-  entity_descriptor: |
-EOF
-    # This needs to be properly indented, hence the sed command
-    printf "$${SAML_ENTITY_DESCRIPTOR}" | sed 's/^/\ \ \ \ /' >> saml.yaml
-    gravity resource create saml.yaml
-    rm saml.yaml
-  fi
-
-  #
   # Only if an opscenter with public access enabled, try to provision DNS / letsencrypt
   # TODO(knisbet) this should be offloaded to an automatic DNS service loaded in k8s / the app
   # But temporarily we'll provision from cloud-init script
-  #
-  # IF the cluster has the gravity-public svc, it must be an externally facing ops center
-  if gravity enter -- --notty /usr/bin/kubectl -- --namespace=kube-system get svc/gravity-public; then
-    #
+  if [ ! -z "${ops_advertise_addr}" ]; then
+    # Manually create a service for OpsCenter, since newer verions of Gravity enable Hub if we enable the
+    # gravity-public service over the installation flag.
+    cat <<EOF | kubectl apply -f -
+apiVersion: v1
+kind: Service
+metadata:
+  annotations:
+    service.beta.kubernetes.io/aws-load-balancer-connection-idle-timeout: "3600"
```

Comment:
Might wanna add the "external-dns" annotation to the service as well.

Reply:
I omitted it here because `external-dns` is not included in Gravity by default. Instead, the DNS setup is done manually over the AWS API below.